packages/ui/src/components/va-form/VaForm.stories.ts

```diff
@@ -1,12 +1,443 @@
+      ref="rulesFormRef"
+      immediate
+      :hideErrorMessages="true"
+    >
+      <va-input
+        v-model="input"
+        label="with no error message"
+        :rules="[value => 'error']"
+      />
+    </va-form>
+    [default]
+    <va-form
+      class="flex max-w-sm"
+      ref="rulesFormRef"
+      immediate
+      :hideErrorMessages="false"
+    >
+      <va-input
+        v-model="input"
+        label="with error message"
+        :rules="[value => 'error']"
+      />
+    </va-form>
+  `,
+})
+
+HideErrorMessages.play = async ({ step }) => {
+  const [firstDefinedInput, secondDefineInput] = document.querySelectorAll('div.va-input-wrapper > div.va-message-list')
+
+  await step('Does not display validation error', async () => {
+    expect(firstDefinedInput).toBeDefined()
+  })
+
+  await step('Displays validation error', async () => {
+    expect(secondDefineInput).toBeUndefined()
+  })
+}
+
+export const HideErrors = () => ({
+  components: { VaForm, VaInput, VaButton },
+  data: () => ({ input: 'error' }),
+  template: `
+    [hidden]
+    <va-form
+      class="flex max-w-sm"
+      immediate
+      :hideErrors="true"
+    >
+      <va-input
+        v-model="input"
+        label="with no error highlight"
+        :rules="[value => 'error']"
+      />
+    </va-form>
+    [default]
+    <va-form
+      class="flex max-w-sm"
+      immediate
+      :hideErrors="false"
+    >
+      <va-input
+        v-model="input"
+        label="with error highlight"
+        :rules="[value => 'error']"
+      />
+    </va-form>
+  `,
+})
+
+HideErrors.play = async ({ step }) => {
+  const [firstInput, secondInput] = document.querySelectorAll('div.va-input-wrapper')
+
+  await step('Does not highlight invalid input', async () => {
+    expect(firstInput).not.toHaveClass('va-input-wrapper--error')
+  })
+
+  await step('Highlights invalid input', async () => {
+    expect(secondInput).toHaveClass('va-input-wrapper--error')
+  })
+}
+
+export const Focus = () => ({
+  components: { VaForm, VaInput, VaButton },
+  template: `
+    <va-form 
+      class="flex space-x-2 mb-6 max-w-sm" 
+      ref="focusForm"
+    >
+      <va-input
+        label="should be focused"
+      />
+      <va-input
+        label="should not be focused"
+      />
+    </va-form>
+    <va-button @click="$refs.focusForm.focus()">
+      Focus first input
+    </va-button>
+  `,
+})
+
+Focus.play = async ({ canvasElement, step }) => {
+  const canvas = within(canvasElement)
+  const firstInput = canvas.getByRole('textbox', { name: 'should be focused' }) as HTMLElement
+  const secondInput = canvas.getByRole('textbox', { name: 'should not be focused' }) as HTMLElement
+  const button = canvas.getByRole('button', { name: 'Focus first input' }) as HTMLElement
+
+  await step('Focusses first input', async () => {
+    await userEvent.click(button)
+    expect(firstInput).toHaveFocus()
+  })
+
+  await step('Does not focus first input', async () => {
```

Comment:
We can remove this interaction. Focus could be only on one element at once. So if it's on first - then it won't be on second.

We also don't need to test for switching focus on subsequent `focus()`, as that's just too hard to implement to be a random regression.